api/app/registration/api.py

```diff
@@ -149,6 +151,20 @@ def post(self):
         if existing_offer:
             return errors.DUPLICATE_OFFER
 
+        existing_outcome = outcome_repository.get_latest_by_user_for_event(user_id, event_id)
+        if existing_outcome:
+            if existing_outcome.status == Status.REJECTED:
+                return errors.CANDIDATE_REJECTED
+            existing_outcome.reset_latest()
+
+        new_outcome = Outcome(
+            event_id,
+            user_id,
+            Status.ACCEPTED,
+            g.current_user['id']
+        )
+        outcome_repository.add(new_outcome)
```

Comment:
Is this going to be used to register a user who was waitlisted? So it's almost like a shortcut instead of setting their status to accepted first?

Reply:
Yes exactly, this ensures that we don't have conflicting state, in one case a user has an offer for a place at the event, but on the other hand the outcome says they're waitlisted. Since we're initially going to be doing this by making direct API calls rather than through a GUI, this will be easier than making a call to outcome API first and then adding the offer :)